runtime-transaction/src/compute_budget_instruction_details.rs

```diff
@@ -0,0 +1,349 @@
+use {
+    solana_compute_budget::compute_budget_limits::*,
+    solana_sdk::{
+        borsh1::try_from_slice_unchecked,
+        compute_budget::{self, ComputeBudgetInstruction},
+        instruction::{CompiledInstruction, InstructionError},
+        pubkey::Pubkey,
+        saturating_add_assign,
+        transaction::{Result, TransactionError},
+    },
+};
+
+#[cfg_attr(test, derive(Eq, PartialEq))]
+#[derive(Default, Debug)]
+pub struct ComputeBudgetInstructionDetails {
```

Comment:
I don't think this raw struct should be public if we can avoid it. We only want the sanitized details to be usable outside this crate right?

Reply:
good call, making all of them private for now. c134a39a3cda19bfca85e49cc6de13b0a1f932b4 May make `InstructionDetails` public when/if needed.